python/lsst/obs/decam/ingestCalibs.py

```diff
@@ -154,18 +167,26 @@ def getDestination(self, butler, info, filename):
         raw : `str`
             Destination filename.
         """
-        # Arbitrarily set ccdnum and calib_hdu to 1 to make the mapper template happy
-        info["ccdnum"] = 1
-        info["calib_hdu"] = 1
         calibType = self.getCalibType(filename)
-        if "flat" in calibType.lower():
-            raw = butler.get("cpFlat_filename", info)[0]
-        elif ("bias" or "zero") in calibType.lower():
-            raw = butler.get("cpBias_filename", info)[0]
-        elif ("illumcor") in calibType.lower():
-            raw = butler.get("cpIllumcor_filename", info)[0]
+        md = readMetadata(filename, self.config.hdu)
+        if not md.exists("PROCTYPE"):
+            raise RuntimeError("Unable to find the calib header keyword PROCTYPE in %s, hdu %d" %
+                               (filename, self.config.hdu))
+        proctype = md.getScalar("PROCTYPE").strip()
+        if "MasterCal" in proctype:
+            # Arbitrarily set ccdnum and calib_hdu to 1 to make the mapper template happy
+            info["ccdnum"] = 1
+            info["calib_hdu"] = 1
+            if "flat" in calibType.lower():
+                raw = butler.get("cpFlat_filename", info)[0]
+            elif ("bias" or "zero") in calibType.lower():
+                raw = butler.get("cpBias_filename", info)[0]
+            elif ("illumcor") in calibType.lower():
+                raw = butler.get("cpIllumcor_filename", info)[0]
+            else:
+                assert False, "Invalid CP calibType '{:s}'".format(calibType)
```

Comment:
`assert False` should be replaced with raising an appropriate exception. `RuntimeError` is a good default, but there may be a better option (ask on slack#dm-butler?). You can also simplify the string to an f-string.